kafka-connect-hive/src/main/scala/com/landoop/streamreactor/connect/hive/sink/HiveSinkTask.scala

```diff
@@ -103,26 +115,34 @@ class HiveSinkTask extends SinkTask with StrictLogging {
     * may be changing, eg, in a rebalance. Therefore, we must commit our open files
     * for those (topic,partitions) to ensure no records are lost.
     */
-  override def close(partitions: util.Collection[KafkaTopicPartition]): Unit = {
+  override def close(partitions: util.Collection[KafkaTopicPartition]): Unit = execute {
     val topicPartitions = partitions.asScala.map(tp => TopicPartition(Topic(tp.topic), tp.partition)).toSet
     close(topicPartitions)
   }
 
   // closes the sinks for the given topic/partition tuples
   def close(partitions: Set[TopicPartition]): Unit = {
     partitions.foreach { tp =>
-      sinks.remove(tp).foreach(_.close)
+      sinks.remove(tp).foreach(_.close())
```

Comment:
wrap in execute to be sure ?

Reply:
this close is called from the override. i made it private the method